discovery-provider/src/queries/query_helpers.py

```diff
@@ -546,8 +547,30 @@ def get_repost_counts(session, query_by_user_flag, query_repost_type_flag, filte
         repost_counts_query = repost_counts_query.filter(
             Repost.blocknumber <= max_block_number
         )
+
+    return repost_counts_query
+
+def get_repost_counts(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types, max_block_number=None):
+    repost_counts_query = get_repost_counts_query(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types)
+    return repost_counts_query.all()
+
+def get_repost_counts_for_past_n_days(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types, days, max_block_number=None):
+    repost_counts_query = get_repost_counts_query(session, query_by_user_flag, query_repost_type_flag, filter_ids, repost_types)
+    repost_counts_query = repost_counts_query.filter(
+                Repost.created_at >= datetime.now() - timedelta(days=days)
+            )
+
     return repost_counts_query.all()
 
+def get_repost_counts_with_time(time, session, filter_ids):
+    switcher = {
+        'day': get_repost_counts_for_past_n_days(session, False, True, filter_ids, None, 1),
```

Comment:
I think you want to use datetrunc for this
https://stackoverflow.com/questions/52952734/how-to-perform-date-trunc-query-in-postgres-using-sqlalchemy